File: Python/Data_Structure/queue.py
# Queue Data_Structure
class Queue():
    def __init__(self, queue_max_size=500000):
        self.head = 0
        self.tail = 0
        self.queue_max_size = queue_max_size
        self.data = []
    
    def enqueue(self, item):
        self.data.append(item)
        self.tail = (self.tail+1) % (self.queue_max_size+1)

    def dequeue(self):
        if self.head == self.tail:
            return -1
        self.head = (self.head+1) % (self.queue_max_size+1)
        return self.data[self.head-1]

#-------------------------------------------------------------------------------#

queue = Queue(10)

print(f'{queue.dequeue()}')

queue.enqueue(5)
queue.enqueue(6)
queue.enqueue(7)

print(f'{queue.dequeue()}')

print(f'{queue.dequeue()}')
print(f'{queue.dequeue()}')
print(f'{queue.dequeue()}')

File: C/Data_Structure/queue.c
#include <stdio.h>
#define QUEUE_MAX_SIZE 500000

// Queue Data_Structure
typedef struct
{
    int head, tail, data[QUEUE_MAX_SIZE+1];
}Queue;

int enqueue(Queue *queue, int item)
{
    if((queue->tail+1) % (QUEUE_MAX_SIZE+1) == queue->head) return -1;
    queue->data[queue->tail] = item;
    queue->tail = (queue->tail+1) % (QUEUE_MAX_SIZE+1);
    return 0;
}

int dequeue(Queue *queue)
{
    if(queue->head == queue->tail) return -1;
    queue->head = (queue->head+1) % (QUEUE_MAX_SIZE+1);
    return queue->data[queue->head-1];
}

//-------------------------------------------------------------------------------//


int main()
{
    Queue queue;

    printf("%d\n", dequeue(&queue));

    enqueue(&queue, 5);
    enqueue(&queue, 6);
    enqueue(&queue, 7);

    printf("%d\n", dequeue(&queue));
    printf("%d\n", dequeue(&queue));
    printf("%d\n", dequeue(&queue));

    printf("%d\n", dequeue(&queue));
    
    return 0;
}